website/content/docs/developers/python/bolts.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Implementing a Python Bolt
+---
+
+Bolts must implement the `Bolt` interface, which has the following methods.
+
+```python
+class Bolt(BaseBolt):
+  def initialize(self, config, context)
+  
+  def process(self, tup)
+```
+
+* The `initialize()` method is called when the bolt is first initialized and
+provides the bolt with the executing environment. It is equivalent to `prepare()` 
+method of the [`IBolt`](/api/com/twitter/heron/api/bolt/IBolt.html) interface in Java.
+Note that you should not override `__init__()` constructor of `Bolt` class
+for initialization of custom variables, since it is used internally by HeronInstance; instead,
+`initialize()` should be used to initialize any custom variables or connections to databases.
+
+* The `process()` method is called to process a single input `tup` of `HeronTuple` type. This method
+is equivalent to `execute()` method of `IBolt` interface in Java. You can use
+`self.emit()` method to emit the result, as described below.
+
+In addition, `BaseBolt` class provides you with the following methods.
+
+```python
+class BaseBolt:
+  def emit(self, tup, stream="default", anchors=None, direct_task=None, need_task_ids=False)
+  def ack(self, tup)
+  def fail(self, tup)
+  
+  @staticmethod
+  def is_tick(tup)
+  
+  def log(self, message, level=None)
+  
+  @classmethod
+  def spec(cls, name=None, inputs=None, par=1, config=None)
+```
+
+* The `emit()` method is to emit a given `tup`, which can be a `list` or `tuple` of 
+any python objects. Unlike the Java implementation, `OutputCollector`
+doesn't exist in the Python implementation.
+
+* The `ack()` method is to indicate that processing of a tuple has succeeded.
+
+* The `fail()` method is to indicate that processing of a tuple has failed.
+
+* The `is_tick()` method returns whether a given `tup` of `HeronTuple` type is a tick tuple.
+
+* The `log()` method is to log an arbitrary message, optionally providing a logging level.
+By default, its logging level is `info`. Log outputs are redirected to the log file
```

Comment:
1. is used to log...
2. It accepts an optional argument which specifies logging level